lib/esp_sdk/configure.rb

```diff
@@ -1,32 +1,36 @@
 module EspSdk
   class Configure
-    attr_accessor :token, :email, :version, :token_expires_at
+    attr_accessor :token, :email, :version, :token_expires_at, :end_point, :domain
 
     def initialize(options)
-      @email   = options[:email]
-      @version = options[:version] || 'v1'
-      token_setup(options)
+      self.email = options[:email]
+      self.version = options[:version] || 'v1'
+      self.domain = options[:domain]
+      self.token = options[:password] || options[:token]
+      self.end_point = options[:password].present? ? 'new' : 'valid'
+      token_setup
     end
 
-    def uri
-      return @uri if @uri.present?
+    def url(endpoint)
```

Comment:
This is a different endpoint. The endpoint set on the class represents the endpoint to use for validating the token. The endpoint passed here represents any endpoint on the api. It also gets called from the Base class.